common/lib/dependabot/notifications.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module Dependabot
```

Comment:
Thoughts on keeping track of event names here as constants?

Reply:
I dig it!

Reply:
Hmm, it does make it harder to remove the instrumentation. When this is just a string, we can keep the `subscribe` call in the calling code while we remove all of the instrumentation code, the calling code will just noop. Now we need to be careful to not remove the constant because it would break upstream usage

Reply:
In this single scenario I don't think ^ matters much, but when thinking about a generic system that lets us instrument all sort of things, I feel like optimizing to make it easy to add/delete notifications without friction is something to strive for. Thoughts?

Reply:
Good point, maybe doesn't make sense to add all notifications as constants but seems ok for ones that we care more about routing to specific places, for this one we'll probably want to end up persisting on update jobs directly whereas other ones we might just want to pass straight through to datadog?

Reply:
I agree that ideally it would be easier to add more, but I'm very mildly paranoid that core, as a public library, could have instrumentation added to it that we might not want to relay through our runner to instrumentation services.

Having a list of constants in Dependabot Core allows us to programmatically filter events as we can, by default relay all constants in Dependabot::Notifications. 

We'd have to consciously bump dependabot-core in our runner to allow new notification to relay which would decouple a core upgrade from a sudden surge of new events hitting our instrumentation layer.

`</tinfoil-hat>`

Reply:
Good point @brrygrdn, my thinking was that we would only subscribe to specific events and not relay anything, but I can definitely see us moving towards that approach